agent/src/agent/pipeline/config/jython_scripts/elastic_loader.py

```diff
@@ -0,0 +1,127 @@
+    if sdc.userParams['INITIAL_TIMESTAMP']:
+        return int(sdc.userParams['INITIAL_TIMESTAMP'])
+    return get_now_with_delay() - get_interval()
+
+
+def get_interval():
+    return int(sdc.userParams['INTERVAL'])
+
+
+def make_request(client, index, params={}, scroll=None):
+    for i in range(1, N_REQUESTS_TRIES + 1):
+        try:
+            page = client.search(
+                body=params,
+                index=index,
+                scroll=scroll,
+                timeout=sdc.userParams['QUERY_TIMEOUT']
+            )
+            sid = page['_scroll_id']
+            scroll_size = page['hits']['total']['value']
+            yield page
+            while scroll_size:
+                page = client.scroll(scroll_id=sid, scroll=scroll)
+                sid = page['_scroll_id']
+                scroll_size = len(page['hits']['hits'])
+                yield page
+        except elasticsearch.exceptions.HTTP_EXCEPTIONS as e:
+            requests.post(sdc.userParams['MONITORING_URL'] + str(e.response.status_code))
+            sdc.log.error(str(e))
+            if i == N_REQUESTS_TRIES:
+                raise
+            time.sleep(i**2)
+            continue
+        break
+
+
+def process_batch(result_, end_):
+    batch = sdc.createBatch()
+    for res in result_['hits']['hits']:
+        record = dict(res.items())
+        record['last_timestamp'] = end_ - get_interval()
+        sdc_record = sdc.createRecord('record created ' + str(get_now_with_delay()))
+        sdc_record.value = record
+        batch.add(sdc_record)
+        if batch.size == sdc.batchSize:
+            batch.process(entityName, str(end_))
+            batch = sdc.createBatch()
+    if batch.size > 0:
+        batch.process(entityName, str(end_))
+
+
+def main():
+    interval = get_interval()
+    end = get_backfill_offset() + interval
+    client = get_client()
+    while True:
+        try:
+            while end > get_now_with_delay():
+                time.sleep(2)
+                if sdc.isStopped():
+                    return
+            if sdc.isStopped():
+                break
+            for batch in make_request(
+                    client,
+                    sdc.userParams['INDEX'],
+                    get_query_params(end),
+                    sdc.userParams.get('SCROLL_TIMEOUT')
+                ):
+                sdc.log.debug(str(batch))
+                if batch['hits']['total']['value']:
+                    process_batch(batch, end)
+                elif sdc.userParams['DVP_ENABLED'] == 'True':
+                    # records with last_timestamp only
+                    batch = sdc.createBatch()
+                    record = sdc.createRecord('record created ' + str(datetime.now()))
+                    record.value = dict()
+                    record.value['last_timestamp'] = end - get_interval()
+                    batch.add(record)
+                    batch.process(entityName, str(end))
+                end += interval
```

Comment:
this shouldn't be inside the loop. in the loop we process data from the same interval